Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -396,16 +441,59 @@ public Apis.Bigquery.v2.Data.Job DoQuery()
         }
 
         /// <summary>
-        /// Copy Job main processing function.
+        /// Copy Job main processing function.  
+        /// *This is written using Apis.BigQuery.v2 becuase Cloud.BigQuery did not 
+        /// support Copy opertations at the time of writing.
         /// </summary>
         public Apis.Bigquery.v2.Data.Job DoCopy()
         {
-            throw new NotImplementedException(
-                "Copy jobs are not implemented yet.  Use the *-BqTabledata cmdlets instead.");
+            if (ShouldProcess($"\n\nCopying table(s) to {Destination.TableId}\n\n"))
+            {
+                try
+                {
+                    var copyjob = new Apis.Bigquery.v2.Data.Job()
+                    {
+                        Configuration = new JobConfiguration()
+                        {
+                            Copy = new JobConfigurationTableCopy()
+                            {
+                                DestinationTable = Destination,
+                                SourceTable = Source,
```

Comment:
There is a `SourceTables` parameter. This may be a case where you want to collect all of the objects in the pipeline, and send them out as one request.